File: src/main/java/com/gtladd/gtladditions/data/recipes/ElectricBlastFurnace.java
package com.gtladd.gtladditions.data.recipes;

import com.gregtechceu.gtceu.api.GTValues;
import com.gregtechceu.gtceu.api.data.tag.TagPrefix;
import com.gregtechceu.gtceu.common.data.GTMaterials;
import com.gregtechceu.gtceu.common.data.GTRecipeTypes;
import net.minecraft.data.recipes.FinishedRecipe;

import java.util.function.Consumer;

public class ElectricBlastFurnace {
    public ElectricBlastFurnace() {}
    public static void init(Consumer<FinishedRecipe> provider) {
        GTRecipeTypes.BLAST_RECIPES.recipeBuilder("magnesium_chloride_dust")
                .inputItems(TagPrefix.dust, GTMaterials.Magnesia).inputItems(TagPrefix.dust, GTMaterials.Carbon)
                .inputFluids(GTMaterials.Chlorine.getFluid(4000))
                .outputItems(TagPrefix.dust, GTMaterials.MagnesiumChloride).outputFluids(GTMaterials.CarbonMonoxide.getFluid(1000))
                .blastFurnaceTemp(2160).EUt(GTValues.VA[GTValues.HV]).duration(200).save(provider);
    }
}

File: src/main/java/com/gtladd/gtladditions/data/recipes/ElectricBlastFurnace.kt
package com.gtladd.gtladditions.data.recipes

import com.gregtechceu.gtceu.api.GTValues
import com.gregtechceu.gtceu.api.data.tag.TagPrefix.dust
import com.gregtechceu.gtceu.common.data.GTMaterials.*
import com.gregtechceu.gtceu.common.data.GTRecipeTypes.BLAST_RECIPES
import com.gtladd.gtladditions.GTLAdditions
import net.minecraft.data.recipes.FinishedRecipe
import java.util.function.Consumer

object ElectricBlastFurnace {
    fun init(provider : Consumer<FinishedRecipe?>) {
        BLAST_RECIPES.recipeBuilder(GTLAdditions.id("magnesium_chloride_dust"))
            .inputItems(dust, Magnesia)
            .inputItems(dust, Carbon)
            .inputFluids(Chlorine.getFluid(4000))
            .outputItems(dust, MagnesiumChloride)
            .outputFluids(CarbonMonoxide.getFluid(1000))
            .blastFurnaceTemp(2160).EUt(GTValues.VA[GTValues.HV].toLong())
            .duration(200).save(provider)
    }
}
